lib/api/LogManagerImpl.cpp

```diff
@@ -749,6 +749,7 @@ namespace MAT_NS_BEGIN
 
     IAuthTokensController* LogManagerImpl::GetAuthTokensController()
     {
+        LOCKGUARD(m_lock);
```

Comment:
Maybe this one is good. Please check if it's `recursive_mutex`, and you'd need to trace the flow where the function is called in the SDK. I honestly don't remember where (it's been 4+ years since I looked into this code).